src/cdk/stepper/step.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  Component, ContentChild, Input, TemplateRef, ViewChild
+} from '@angular/core';
+import {CdkStepLabel} from './step-label';
+import {coerceBooleanProperty} from '../coercion/boolean-property';
+
+@Component({
+  selector: '[cdk-step]',
+  templateUrl: 'step.html',
+})
+export class CdkStep {
+  /** Template for step label if it exists. */
+  @ContentChild(CdkStepLabel) stepLabel: CdkStepLabel;
+
+  /** Template for step content. */
+  @ViewChild(TemplateRef) content: TemplateRef<any>;
+
+  /** Label of the step. */
+  @Input()
+  label: string;
```

Comment:
Tab checks the same way as it's done for the stepper in stepper-horizontal.html and stepper-vertical.html. If `stepLabel` exists, it will use the template; if `stepLabel` does not exist, it will use the text label.